src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseFeedbackCreationService.java

```diff
@@ -194,18 +209,25 @@ public List<Feedback> createFeedbackFromStaticCodeAnalysisReports(List<StaticCod
 
                 // Store static code analysis in JSON format
                 try {
-                    // the feedback is already pre-truncated to fit, it should not be shortened further
-                    feedback.setDetailTextTruncated(mapper.writeValueAsString(issue));
+                    feedback.setDetailTextTruncated(mapper.writeValueAsString(updatedIssue));
                 }
                 catch (JsonProcessingException e) {
-                    continue;
+                    continue;  // Skip this feedback if JSON processing fails
                 }
                 feedbackList.add(feedback);
             }
         }
         return feedbackList;
     }
 
+    private String truncateMessage(String message) {
+        if (message != null) {
+            int maxLength = Math.min(message.length(), FEEDBACK_DETAIL_TEXT_DATABASE_MAX_LENGTH - 500);
```

Comment:
Why the `- 500`?

Reply:
To give space for the JSON structure which will be saved as a string.